### ISSUE TITLE
[Apache Tomcat] System test is failing while TSDB is enabled in package

### ISSUE THREAD

**Issue by milan-elastic** [elastic/integrations#6099]
# Description
System test is getting failed while TSDB is enabled, few fields are expected as `Array` but it is coming as a `String`.

**Steps to reproduce the issue:**

In Manifest files keep the following piece of code to make sure TSDB is enabled:

```
elasticsearch:
  index_mode: "time_series"

```
After making the above changes in `manifest.yml` run the system tests for the package. 
The following error will be generated:

![image (1)](https://user-images.githubusercontent.com/118723373/236218132-0e3e4274-9f55-4635-8320-7b9cb33644de.png)

If the above-mentioned code is removed from the Manifest.yml then the system tests are getting passed.

Currently, in none of the integration this error is getting generated because TSDB is not by default enabled in the datastream's manifest.yml of shipped package. 
But in Tomcat integration we are planning to keep the TSDB by default enabled hence we are facing this issue, and there are high chance that other integrations also face the same issue when the TSDB will be enabled by default. 

While debugging more into it we found that by keeping the TSDB enabled some fields are coming in String format. but actually, it should be Array only. Once we remove the default TSDB enablement then these fields are coming as Array(which is expected). Please find the Tomcat examples below.

**Event Before TSDB enabled:**
 ```
{
"host": {
"ip": [
        "127.0.0.1"
      ]
"mac": [
        "02-42-C1-A6-10-09"
      ],
}
"event": {
 "type": [
        "info"
      ],
      "category": [
        "web"
      ],
}
}

```
**Event After TSDB enabled:**
```
{
"event": {
 "category": "web",
 "type": "info",
},
"host":{
"ip": "127.0.0.1",
 "mac": "02-42-C1-A6-10-09",
}
}
```

**Comment by agithomas** [comment]
Can you please verify if you are using the latest version of elastic-package?

**Comment by agithomas** [comment]
Please mention the name of the integration you are facing the problem ?

**Comment by felixbarny** [comment]
What's the test condition that results in this error?

It seems we might be doing assertions on the `_source` of the documents and I'm wondering why we need to do that. With synthetic _source, it's expected that the source doesn't look exactly like the original source. However, the _source shouldn't be relevant for metrics use cases, as you're typically not looking at individual documents but rather do aggregations over many documents. Aggregations are performed over the columnar doc_values data structure which doesn't change with or without TSDB (aside from a more efficient encoding). Internally in ES, all files may be multi-valued. 

If we want do do assertions on certain fields, we should get those values from [`fields`](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-fields.html#search-fields-param) instead of `_source`.

**Comment by milan-elastic** [comment]
> Can you please verify if you are using the latest version of elastic-package? If not, please test using the latest version of elastic-package

Yes, we have tested with elastic-package `v0.79.0` and `kibana v8.7.0 || v8.8.0-SNAPSHOT` and still facing the issue.

**Comment by milan-elastic** [comment]
> Please mention the name of the integration / data stream you are facing the problem ? Is the problem affecting all the datastreams or just one?

For now, 2 (cache and request) datastreams we are facing this issue in Apache Tomcat integration because others are logs datastreams so for that we don't need to keep the TSDB enabled.

**Comment by agithomas** [comment]
> What's the test condition that results in this error?
> 
> It seems we might be doing assertions on the `_source` of the documents and I'm wondering why we need to do that. With synthetic _source, it's expected that the source doesn't look exactly like the original source. However, the _source shouldn't be relevant for metrics use cases, as you're typically not looking at individual documents but rather do aggregations over many documents. Aggregations are performed over the columnar doc_values data structure which doesn't change with or without TSDB (aside from a more efficient encoding). Internally in ES, all files may be multi-valued.
> 
> If we want do do assertions on certain fields, we should get those values from [`fields`](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-fields.html#search-fields-param) instead of `_source`.

@jsoriano , can you please comment on the behaviour of `system test` to adjust with the synthetic source, enabled on TSDB enablement ?

**Comment by lalit-satapathy** [comment]
_index_mode: "time_series"_ enablement does not need to go now. If needed and can go later after the PRs are merged.

**Comment by jsoriano** [comment]
> > If we want do do assertions on certain fields, we should get those values from [`fields`](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-fields.html#search-fields-param) instead of `_source`.
> 
> @jsoriano , can you please comment on the behaviour of `system test` to adjust with the synthetic source, enabled on TSDB enablement ?

I guess that we are doing this for historical reasons, this is what used to be done in Beats, and it used to be enough.

We have also identified a related issue when thinking about support for runtime fields (see https://github.com/elastic/elastic-package/issues/1229).

I have created a new issue specific to synthetics source support (https://github.com/elastic/elastic-package/issues/1253). We may end up basing the validation on `fields` to solve these issues.

**Comment by ruflin** [comment]
There is an urgency to get this fixed as it holds us back from enabling TSDB on data streams. This looks like the proper long term fix: https://github.com/elastic/elastic-package/issues/1253 If this is not quickly possible, maybe we can just to all the comparison assuming each field is an array? So if we see `host.ip` not being array, we convert it for comparison to an array with one element.

**Comment by jlind23** [comment]
@milan-elastic now that @mrodm fixed https://github.com/elastic/elastic-package/issues/1253 you should be able to move forward here.
cc @andresrc

**Comment by agithomas** [comment]
@milan-elastic , can you upgrade the elastic-package and verify the system test run with TSDB enabled?

**Comment by milan-elastic** [comment]
> @milan-elastic , can you upgrade the elastic-package and verify the system test run with TSDB enabled?

@agithomas I tried in latest elastic-package 0.81.0, but in this version system testcases are not getting passed. I am getting following error. 

![image](https://github.com/elastic/integrations/assets/118723373/00848407-d8b6-48f6-b466-25bee9582807)

May be next version of elastic-package will have fix for this error. I found one [issue](https://github.com/elastic/elastic-package/issues/1296) related to it.

**Comment by milan-elastic** [comment]
> > May be next version of elastic-package will have fix for this error. I found one [issue](https://github.com/elastic/elastic-package/issues/1296) related to it.
> 
> @milan-elastic Yes, there was an issue with that version when it is used docker-compose v1. Next elastic-package version will contain the fix for that.
> 
> In the meantime, as that PR needed for that issue was already merged, you can try to test this locally by building your own `elastic-package` from the latest changeset in main branch https://github.com/elastic/elastic-package#development

I just checked with the latest elastic-package version (0.82.0). With TSDB enabled, system testcases are getting passed now. cc: @agithomas

**Comment by andresrc** [comment]
@agithomas @milan-elastic @mrodm can this issue be closed, then?

**Comment by agithomas** [comment]
Closing the issue as the team is unblocked and create system test
